drivers/flash/nrf_qspi_nor.c

```diff
@@ -1398,38 +1409,34 @@ static int qspi_nor_pm_action(const struct device *dev,
 }
 #endif /* CONFIG_PM_DEVICE */
 
-void  z_impl_nrf_qspi_nor_base_clock_div_force(const struct device *dev,
-					       bool force)
+void z_impl_nrf_qspi_nor_xip_enable(const struct device *dev, bool enable)
 {
-#if defined(CONFIG_SOC_SERIES_NRF53X)
 	struct qspi_nor_data *dev_data = dev->data;
-	/*
-	 * The divider is normally changed, unless the flag is set, only for
-	 * periods when the driver is locked, so the flag itself also can only
-	 * be modified while the driver is locked.
-	 */
+
+	qspi_device_init(dev);
+
+#if NRF_QSPI_HAS_XIPEN
+	nrf_qspi_xip_set(NRF_QSPI, enable);
+#endif
 	qspi_lock(dev);
-	dev_data->keep_base_clock_div_set = force;
+	dev_data->xip_enabled = enable;
 	qspi_unlock(dev);
-#else
-	ARG_UNUSED(dev);
-	ARG_UNUSED(force);
-#endif
+
+	qspi_device_uninit(dev);
```

Comment:
Does this entire sequence be called for enabling the XIP for non-nrf53 devices then?
I think it should be doing nothing for other nRF devices and the initialization of `qspi_nor_dev_data` structure should just assign true to `xip_enabled`, and the function could be empty.
I am wondering whether we can get customer issue where somebody will observe `xip_enabled` being changed by the function, yet nothing happening.

Reply:
> I think it should be doing nothing for other nRF devices and the initialization of `qspi_nor_dev_data` structure should just assign true to `xip_enabled`, and the function could be empty.

No, the `xip_enabled` flag is intentionally not set to `true` by default:
https://github.com/zephyrproject-rtos/zephyr/blob/b8265d5e06a7c23437f9f3c7a870b8a687cbe800/include/zephyr/drivers/flash/nrf_qspi_nor.h#L17-L26

Reply:
Sorry, I missunderstood the function, by `enable` suggesting that the feature is enabled/disabled, rather than in-use as it really is.